riptide-core/src/test/java/org/zalando/riptide/FailedDispatchTest.java

```diff
@@ -139,9 +137,12 @@ public void shouldThrowOnFailedConversionBecauseOfFaultyBody() {
 
     @Test
     public void shouldHandleNoBodyAtAll() {
+        final HttpHeaders headers = new HttpHeaders();
+        headers.setContentLength(0);
```

Comment:
Spring changed its heuristic to detect an empty body somewhere between 4.1 and 4.3. Before it checked for 204 No Content and the Content-Length. In the later versions it also peeks at the first byte of the body.